quiche/src/recovery/mod.rs

```diff
@@ -321,37 +322,33 @@ impl Recovery {
 
     pub fn set_pacing_rate(&mut self, rate: u64) {
         if rate != 0 {
+            self.pacer.update(self.send_quantum, rate);
+
             self.pacing_rate = rate;
         }
     }
 
     pub fn get_packet_send_time(&self) -> Instant {
-        self.last_packet_scheduled_time
+        self.pacer.next_time()
     }
 
     fn schedule_next_packet(
         &mut self, epoch: packet::Epoch, now: Instant, packet_size: usize,
     ) {
         // Don't pace in any of these cases:
         //   * Packet epoch is not EPOCH_APPLICATION.
-        //   * Packet contains only ACK frames.
-        //   * The start of the connection.
-        if epoch != packet::EPOCH_APPLICATION ||
-            packet_size == 0 ||
-            self.bytes_sent <= self.congestion_window ||
+        //   * Packet contains has no data.
```

Comment:
It looks like you removed the `packet_size == 0` check, so this comment is not true anymore?

But also, typo :)

```suggestion
        //   * Packet contains no data.
```